Sources/Tracing/Docs.docc/Guides/TraceYourApplication.md

```diff
@@ -0,0 +1,443 @@
+
+This will allow not only your code, that we're about to write, to use tracing, but also all other libraries which
+have been implemented against the distributed tracing API to use it as well. For example, by configuring the global
+tracing system, an HTTP server or client will automatically handle trace propagation for you, so make sure to always
+bootstrap your tracer globally, otherwise you might miss out on its crucial context propagation features.
+
+How the tracer library is initialized will differ from library to library, so refer to the respective implementation's
+documentation. Once you're ready, pass the tracer or instrument to the ``InstrumentationSystem/bootstrap(_:)`` method, 
+e.g. like this:
+
+```swift
+import Tracing // this library
+
+// Import and prepare whatever the specific tracer implementation needs.
+// In our example case, we'll prepare the OpenTelemetry tracing system:
+import NIO
+import OpenTelemetry
+
+let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+let otel = OTel(serviceName: "onboarding", eventLoopGroup: group)
+try otel.start().wait()
+
+// Bootstrap the tracing system:
+InstrumentationSystem.bootstrap(otel.tracer())
+```
+
+You'll notice that the API specifically talks about Instrumentation rather than just Tracing.
+This is because it is also possible to use various instrumentation systems, e.g. which only take care
+of propagating certain `Baggage` values across process boundaries, without using tracing itself.
+
+In other words, all tracers are instruments, and the `InstrumentationSystem` works equally for ``InstrumentProtocol``,
+as well as ``TracerProtocol`` implementations.
+
+Our guide focuses on tracing through, so let's continue with that in mind.
+
+#### Recommended bootstrap order
+
+Swift offers developers a suite of observability libraries: logging, metrics and tracing. Each of those systems offers a `bootstrap` function. It is useful to stick to a recommended boot order in order to achieve predictable initialization of applications and sub-systems.
+
+Specifically, it is recommended to bootstrap systems in the following order:
+
+1. [Swift Log](https://github.com/apple/swift-log#default-logger-behavior)'s `LoggingSystem`
+2. [Swift Metrics](https://github.com/apple/swift-metrics#selecting-a-metrics-backend-implementation-applications-only)' `MetricsSystem`
+3. [Swift Distributed Tracing](https://github.com/apple/swift-distributed-tracing)'s `InstrumentationSystem`
+4. Finally, any other parts of your application
+
+This is because tracing systems may attempt to emit logs or metrics about their status etc.
+
+If you intend to use trace identifiers for log correlation (i.e. logging a `trace-id` in every log statement that is part of a trace),
+then don't forget to also configure a swift-lot `MetadataProvider`.
+
+A typical bootstrap could look something like this:
+
+```swift
```

Comment:
Maybe keep this as a full example (with MultiThreadedEventLoopGroup)  and remove the bootstrapping the tracer above since this is a more complete example and having the two very close to each other is duplication?

Reply:
And also add a note about `BatchSpanProcessor` vs `SimpleSpanProcessor`. As simple processor is not suitable for hot code execution as will lead to 700+ byte message being sent each time the span is executed.

Reply:
I'm a bit torn about this... This isn't the otel tracer's guide. For what it's worth this should be "go and read a specific tracer impls docs" but I want something good enough here to get someone started.

I'll see how to improve this bit though